### ISSUE TITLE
/036-custom-input-control/ invalidates too often on Edge

### ISSUE THREAD

**Issue by bborgesr** [rstudio/shiny#1637]
I'm pretty sure I'm using the word "invalidate" wrong in here. The issue is that on Edge, this app (https://winston.shinyapps.io/036-custom-input-control/) looks like:

![screen shot 2017-03-30 at 2 18 37 am](https://cloud.githubusercontent.com/assets/6527540/24483357/7e948f7a-14ef-11e7-8257-d0747f9be153.png)

Whereas in Chrome, it looks like:

![screen shot 2017-03-30 at 2 18 59 am](https://cloud.githubusercontent.com/assets/6527540/24483364/8b6d4282-14ef-11e7-84d9-69b3b66f9c61.png)

I.e. in Edge, whenever we change states from one column to the other, the verbatimTextOutput changes correctly but then immediately "invalidates" until we make a change (which results in the same thing - it comes back to life, updates correctly, and immediately invalidates again)

**Comment by bborgesr** [comment]
same issue here: https://winston.shinyapps.io/098-plot-interaction-article-3/:

![screen shot 2017-03-30 at 3 16 03 am](https://cloud.githubusercontent.com/assets/6527540/24484653/3f428cca-14f7-11e7-8981-b0ea79de3bff.png)

**Comment by wch** [comment]
Very strange. I just tested 098-plot-interaction-article-3 locally with Edge, and I can get the same problem. Even though it grays out, it still seems to work.

**Comment by bborgesr** [comment]
Yeah, it happened for a bunch of apps... After those two, I stopped making a note of it, since it's obviously the same bug and either of those can serve as a repro. It's not a terrible bug, because it does work "correctly," but it's definitely strange...

**Comment by wch** [comment]
I think this may be a problem in Edge itself. What I see happening is that the `recalculating` class gets added to the text output and then removed very quickly. But sometimes the browser doesn't seem to update the appearance when the class is removed. This happens in maybe 1 out of 10 cases for me.

**Comment by wch** [comment]
I think this has to do with how Edge handles CSS transitions. If you load https://winston.shinyapps.io/098-plot-interaction-article-3/ and run this code at the JS in other browsers, no visible changes will happen to the plot. This code adds the `recalculating` class, then removes it in 200ms. In the CSS rule for `.recalculating`, there's a 250ms delay before it fades to gray, but before that happens, the class gets removed:

```JS
$('#plot1').addClass('recalculating');
setTimeout(function() { $('#plot1').removeClass('recalculating'); }, 200);
```

However, in Edge, what happens is that it grays out and stays that way. So with Edge, when you add a CSS class with a transition, it looks like it will do the transition even if the class is removed before the transition starts. I'm not sure if this is a bug or a feature.

**Comment by wch** [comment]
This is my attempt to reproduce the weird behavior in Edge with a simplified example, but it doesn't seem to work: https://jsfiddle.net/winstonchang/pbbnLm4o/

**Comment by jcheng5** [comment]
With your jsfiddle example, it does the transition and stays transparent iff I don't move my mouse off the button.

**Comment by jcheng5** [comment]
@bborgesr Are all the apps that repro'd this, ones that you interact with a plot (click or brush)?

**Comment by bborgesr** [comment]
@jcheng5: no, for example, [the first app I mentioned](https://winston.shinyapps.io/036-custom-input-control/) (see issue description) is just moving things around... Let me see if your fix (#1640) solves it

**Comment by bborgesr** [comment]
I can confirm it's fixed with #1640 (at least for these two cases)!

See:

- https://barbara.shinyapps.io/036-custom-input-control/

- https://barbara.shinyapps.io/098-plot-interaction-article-3/
